Codes/Visualization/ConstructOrderBookVisualization_20200214.R

```diff
@@ -20,11 +20,27 @@ library(reshape2)
 #1: Buy limit order
 #Note: Execution of a sell (buy) limit order corresponds to a buyer (seller) initiated trade, i.e. buy (sell) trade.
 
+direc<-"C:/DatiLocali/Dropbox/Trader Anonymity/Data/RawData/"
+stock<-"AAPL"
+day<-"04"
+
 ##READ IN RAW DATA
-file<-"C:/DatiLocali/Dropbox/Trader Anonymity/Data/RawData/AAPL_2013-11-04_34200000_59400000_message_10.csv"
+file<-paste0(direc,stock,"_2013-11-",day,"_34200000_59400000_message_10.csv")
 cols<-c("time","eventtype","orderID","size","price","direction","mpid")
 mydata<-read.csv(file=file,header=F,col.names=cols,stringsAsFactors=F)
 rm(file,cols)
+##READ IN RAW DATA
+file<-paste0(direc,stock,"_2013-11-",day,"_34200000_59400000_orderbook_10.csv")
+Cols<-c("askprice1","asksize1","bidprice1","bidsize1","askprice2","asksize2","bidprice2","bidsize2",
+        "askprice3","asksize3","bidprice3","bidsize3","askprice4","asksize4","bidprice4","bidsize4",
+        "askprice5","asksize5","bidprice5","bidsize5","askprice6","asksize6","bidprice6","bidsize6",
+        "askprice7","asksize7","bidprice7","bidsize7","askprice8","asksize8","bidprice8","bidsize8",
+        "askprice9","asksize9","bidprice9","bidsize9","askprice10","asksize10","bidprice10","bidsize10")
+mydata2<-read.csv(file=file,header=F,col.names=Cols,stringsAsFactors=F)
+rm(file)
+mydata2[,seq(1,40,2)]<-mydata2[,seq(1,40,2)]/10000
+mydata<-cbind(mydata,mydata2)
+rm(mydata2)
 
 ##REMOVE EXECUTION OF HIDDEN ORDERS
```

Comment:
Since we are only interested in plotting the evolution of visible submissions across the trading day, if I understand correctly the hidden order executions do not contain any useful information. There is no way to connect them back to a submission, since the submissons themselves are not visible.

Reply:
Understand. I believe hidden order execution has not impact in the visualization because you re-sample the data in regular time interval. This is not crucial.